### ISSUE TITLE
Double appointments

### ISSUE THREAD

**Issue by Wikibear** [u01jmg3/ics-parser#102]
Loaded Dev-Master today get one appointments twice. 

```
<?php
	$path = "";
	$filedata = "";
	require $path.'class/ICal/ICal.php';
	require $path.'class/ICal/EventObject.php';
	use ICal\ICal;
	$ical = new ICal("https://www.google.com/calendar/ical/um8r7ne385gpag3mla4undh2u0%40group.calendar.google.com/public/basic.ics");

	function nl2br2($string) {
		$string = str_replace(array("\\n", "\\, "), "<br />", $string);
		$string = preg_replace('@(https?://([-\w\.]+[-\w])+(:\d+)?(/([\w/_\.#-]*(\?\S+)?[^\.\s])?)?)@', '<a href="$1" target="_blank">$1</a>', $string);
		return $string;
	} 

	function template($summary,$start,$end,$description,$location){
		$filedatareturn = "<p align=\"left\"><b>Was:".nl2br2($summary)."</b><br />";
		$filedatareturn .= "Wann: ".$start."<br />";
		$filedatareturn .= "Wo: ".nl2br2($location)."<br />";
		$filedatareturn .= "Beschreibung: ".nl2br2($description)."</p><hr />";
		return $filedatareturn;
	}
	$events = $ical->events();
	$events = $ical->eventsFromRange(date("Y-m-d"), date("Y-m-t"));
	$i=1;
	foreach($events as $event){
		echo $i;
		$filedata .= template($event->summary,date('d.m.Y H:i', $ical->iCalDateToUnixTimestamp($event->dtstart)),date('H:i', $ical->iCalDateToUnixTimestamp($event->dtend)),$event->description,$event->location);
		$i++;
	}
	echo $filedata;
?>
```

Get as result:

![2017_01_22_11_03_51_mozilla_firefox](https://cloud.githubusercontent.com/assets/18314278/22181754/c1e1ccac-e092-11e6-9a21-32c97a25ff75.png)

But there is only one date by LCBS in google kalendar as you can see here:

![2017_01_22_11_05_21_google_kalender_woche_vom_23 _jan _2017](https://cloud.githubusercontent.com/assets/18314278/22181761/d7a8babe-e092-11e6-9a10-aad5891160c5.png)

Why I get this date twice?

Seems to be a google error. There are a hidden date which will not show up in kalendar. I have different UIDS. Thats mystery... I will take an investigation in this.

**Comment by u01jmg3** [comment]
If you want me to investigate anything, I will need the iCal event that produces the problem.

**Comment by Wikibear** [comment]
Found the issue! It's Google dependent issue. This date is every last Friday. If you want to get the last Friday you must look for a month that is near the end of a month. This is not February. Google added two dates and show only one. I deleted this date, but this date will stay. I cycle thru each month and found a date with same data, but start very late in the year. So I don't know was happening there. I deleted this date also and all dates are gone. Now start with a new date and this new date are fine. This is crazy. I look into the code all time long and Google makes this issue. I found this cause of different UIDS. 

Thanks for your support! This issue can be closed.
